### DISCUSSION TITLE
How to use FieldMasks?

### DISCUSSION THREAD

**Discussion by banool** [hyperium/tonic#1324] [Q&A]
Hi, I'm interested in using FieldMasks for reads (I have no use for them for writes right now). I see they exist in [prost_types](https://docs.rs/prost-types/latest/prost_types/struct.FieldMask.html)`, but I don't see them referred to anywhere in tonic:
```
$ rg -i fieldmask
examples/proto/googleapis/google/pubsub/v1/pubsub.proto
261:  google.protobuf.FieldMask update_mask = 2
919:  google.protobuf.FieldMask update_mask = 2
1192:  google.protobuf.FieldMask update_mask = 2

$ rg -i field_mask
examples/proto/googleapis/google/pubsub/v1/pubsub.proto
25:import "google/protobuf/field_mask.proto";

tests/wellknown-compiled/proto/google.proto
10:import "google/protobuf/field_mask.proto";
```

Are these supported somehow? I imagine it should be possible to access it as part of the `Request` object wrapping the request message.

I searched the issues and discussions here and found now questions about them so far.

**Comment by LucioFranco** [comment]
No there is no support as far as I know

**Reply by banool** [reply]
Okay great, thanks for letting me know.

**Comment by frederikhors** [comment]
@banool did you find a way?

**Reply by banool** [reply]
Unfortunately not, I just don't think it is really possible with the library support we have today in Rust.

**Comment by akselerando** [comment]
@LucioFranco Are there any plans for supporting FieldMasks in the (near) future? 
What are the best alternatives?

**Reply by LucioFranco** [reply]
Not at the moment.
